usr/local/www/services_dhcpv6.php

```diff
@@ -371,552 +376,612 @@
-		document.iform.maxtime.disabled = endis;
-		//document.iform.gateway.disabled = endis;
-		document.iform.dhcpv6leaseinlocaltime.disabled = endis;
-		document.iform.domain.disabled = endis;
-		document.iform.domainsearchlist.disabled = endis;
-		document.iform.ddnsdomain.disabled = endis;
-		document.iform.ddnsdomainprimary.disabled = endis;
-		document.iform.ddnsdomainkeyname.disabled = endis;
-		document.iform.ddnsdomainkey.disabled = endis;
-		document.iform.ddnsupdate.disabled = endis;
-		document.iform.ntp1.disabled = endis;
-		document.iform.ntp2.disabled = endis;
-		//document.iform.tftp.disabled = endis;
-		document.iform.ldap.disabled = endis;
-		document.iform.netboot.disabled = endis;
-		document.iform.bootfile_url.disabled = endis;
-	}
+if (is_subsystem_dirty('staticmaps'))
+	print_info_box_np(gettext('The static mapping configuration has been changed') . '.<br />' . gettext('You must apply the changes in order for them to take effect.'));
 
-	function show_shownumbervalue() {
-		document.getElementById("shownumbervaluebox").innerHTML='';
-		aodiv = document.getElementById('shownumbervalue');
-		aodiv.style.display = "block";
+/* active tabs */
+$tab_array = array();
+$tabscounter = 0;
+$i = 0;
+foreach ($iflist as $ifent => $ifname) {
+	$oc = $config['interfaces'][$ifent];
+	if ((is_array($config['dhcpdv6'][$ifent]) && !isset($config['dhcpdv6'][$ifent]['enable']) && !(is_ipaddrv6($oc['ipaddrv6']) && (!is_linklocal($oc['ipaddrv6'])))) ||
+		(!is_array($config['dhcpdv6'][$ifent]) && !(is_ipaddrv6($oc['ipaddrv6']) && (!is_linklocal($oc['ipaddrv6'])))))
+		continue;
+
+	if ($ifent == $if)
+		$active = true;
+	else
+		$active = false;
+
+	$tab_array[] = array($ifname, $active, "services_dhcpv6.php?if={$ifent}");
+	$tabscounter++;
+}
+/* tack on PPPoE or PPtP servers here */
+/* pppoe server */
+if (is_array($config['pppoes']['pppoe'])) {
+	foreach($config['pppoes']['pppoe'] as $pppoe) {
+		if ($pppoe['mode'] == "server") {
+			$ifent = "poes". $pppoe['pppoeid'];
+			$ifname = strtoupper($ifent);
+
+			if ($ifent == $if)
+				$active = true;
+			else
+				$active = false;
+
+			$tab_array[] = array($ifname, $active, "services_dhcpv6.php?if={$ifent}");
+			$tabscounter++;
+		}
 	}
+}
 
-	function show_ddns_config() {
-		document.getElementById("showddnsbox").innerHTML='';
-		aodiv = document.getElementById('showddns');
-		aodiv.style.display = "block";
-	}
-	function show_ntp_config() {
-		document.getElementById("showntpbox").innerHTML='';
-		aodiv = document.getElementById('showntp');
-		aodiv.style.display = "block";
-	}
-	/*
-	function show_tftp_config() {
-		document.getElementById("showtftpbox").innerHTML='';
-		aodiv = document.getElementById('showtftp');
-		aodiv.style.display = "block";
-	}
-	*/
-	function show_ldap_config() {
-		document.getElementById("showldapbox").innerHTML='';
-		aodiv = document.getElementById('showldap');
-		aodiv.style.display = "block";
-	}
+if (false && $tabscounter == 0) {
```

Comment:
It was leftover Debug. Thanks.

No idea about $tabcounter == 0. That’s the way page was originally coded. I have replaced per your suggestion.

> On Jun 7, 2015, at 4:25 AM, SjonHortensius <notifications@github.com> wrote:
> 
> In usr/local/www/services_dhcpv6.php <https://github.com/SjonHortensius/pfsense/pull/221#discussion_r31874909>:
> 
> > -		aodiv = document.getElementById('showntp');
> > -		aodiv.style.display = "block";
> > -	}
> > -	/*
> > -	function show_tftp_config() {
> > -		document.getElementById("showtftpbox").innerHTML='';
> > -		aodiv = document.getElementById('showtftp');
> > -		aodiv.style.display = "block";
> > -	}
> > -	*/
> > -	function show_ldap_config() {
> > -		document.getElementById("showldapbox").innerHTML='';
> > -		aodiv = document.getElementById('showldap');
> > -		aodiv.style.display = "block";
> > -	}
> > +if (false && $tabscounter == 0) {
> Is this a left-over debug statement? Also, why do you need $tabscounter, is empty($tabs_array); not good enough
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/SjonHortensius/pfsense/pull/221/files#r31874909>.
> 

Steve
--
Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com <mailto:sbeaver@netgate.com> | www.netgate.com <http://www.netgate.com/>